PhotoBooth/Camera Page/CameraViewController.swift

```diff
@@ -187,11 +236,63 @@ class CameraViewController: UIViewController {
             presentCancelationPartialModal()
         }
     }
+
+    @objc func handleSingleTap(tapGesture: UITapGestureRecognizer) {
+        let point = tapGesture.location(in: self.previewLayerContainer)
+        print("current point: \(point)")
+        viewModel.captureSession.focus(touchLocation: point)
+    }
+
+    @objc func handleDoubleTap() {
+        viewModel.captureSession.switchCamera()
+    }
+
+    @objc func pinch(_ pinch: UIPinchGestureRecognizer) {
+        let newScaleFactor = viewModel.captureSession.minMaxZoom(pinch.scale * viewModel.zoomFactor)
+
+        switch pinch.state {
+        case .began: fallthrough
+        case .changed: viewModel.captureSession.updateZoomScaleFactor(scale: newScaleFactor)
+        case .ended:
+            viewModel.updateZoomFactor(factor: newScaleFactor)
+            viewModel.captureSession.updateZoomScaleFactor(scale: viewModel.zoomFactor)
```

Comment:
if the capture session updates it zoom factor does the view model also need to track it?

Reply:
yea the factor here is for the views zoom factor so we can see the visible change and the capture session method sets it so the photo will be captured with the right zoom scale

Reply:
Right but do we need both? In other words why does the view model need to know the zoom scale if the capture session already has it?

Reply:
I'm running into a problem. I can remove all of that but I can't get the zoom scale to go back to 1 after the shoot. the pinch was the thing modifying the scale but idk another way to set it back

Reply:
Do we need it to go back after the shoot? Also Im not sure I understand what you mean by the pinch was the thing modifying the scale. Like I know that pinch modifies the scale but what does that have to do with after the shoot? I thought that was part of the reset method in the view model

Reply:
no the reset can't modify the visible scale